android/src/main/java/com/brentvatne/common/toolbox/AudioManagerDelegate.kt

```diff
@@ -0,0 +1,106 @@
+package com.brentvatne.common.toolbox
+
+import android.content.Context
+import android.media.AudioManager
+import android.media.AudioManager.OnAudioFocusChangeListener
+import com.brentvatne.common.api.RNVPlayerInterface
+import com.facebook.react.uimanager.ThemedReactContext
+
+/**
+ * Delegate audio management to this class
+ * This is an helper to group all generic android code which do not depend on player implementation
+ */
+class AudioManagerDelegate(player: RNVPlayerInterface, themedReactContext: ThemedReactContext) {
```

Comment:
We can add `private val` because they are not expected to change after initialization.
```suggestion
class AudioManagerDelegate(private val player: RNVPlayerInterface, private val themedReactContext: ThemedReactContext
)```